isis/src/mex/apps/hrsc2isis/main.cpp

```diff
@@ -11,93 +11,154 @@
-  outCube = new Cube();
 
-  outCube->setByteOrder(outAtt.byteOrder());
-  outCube->setFormat(outAtt.fileFormat());
-  outCube->setLabelsAttached(outAtt.labelAttachment() == AttachedLabel);
+}
 
-  /**
-    * Isis2 mex2isis.pl:
-    *   if (index($detector_id,"MEX_HRSC_SRC") < 0 &&
-    *   $processing_level_id < 3)
-    */
 
-  bool isSrcFile = (label["DETECTOR_ID"][0] != "MEX_HRSC_SRC");
-  bool mapProjRdr = ((int)label["PROCESSING_LEVEL_ID"] >= 3);
+// Import a PDS3, HRSC, SRC Camera image.
+void ImportHrscSrcImage(ProcessImportPds &p, Pvl &label) {
 
-  try {
-    if (!isSrcFile) {
-      QString msg = "File [" + ui.GetFileName("FROM");
-      msg += "] is SRC data and cannot be read.";
-      throw IException(IException::User, msg, _FILEINFO_);
-    }
+  UserInterface &ui = Application::GetUserInterface();
 
-    if (mapProjRdr) {
-      QString msg = "File [" + ui.GetFileName("FROM");
-      msg += "] is map projected and cannot be read.";
-      throw IException(IException::User, msg, _FILEINFO_);
-    }
+  outCube = p.SetOutputCube("TO");
+  p.StartProcess();
+
+  Pvl otherLabels;
+  TranslateHrscLabels(label, otherLabels);
+
+  std::cout << "========================================" <<
+    otherLabels << "=======================================================" << std::endl;
+
+  if (otherLabels.hasGroup("Instrument") &&
+      (otherLabels.findGroup("Instrument").keywords() > 0)) {
+    outCube->putGroup(otherLabels.findGroup("Instrument"));
   }
 
-  catch (IException &e) {
-      QString msg = "File cannot be read by hrsc2isis, use pds2isis.";
-      throw IException(e, IException::User, msg, _FILEINFO_);
+  if (otherLabels.hasGroup("BandBin") &&
+      (otherLabels.findGroup("BandBin").keywords() > 0)) {
+    outCube->putGroup(otherLabels.findGroup("BandBin"));
   }
 
+  if (otherLabels.hasGroup("Archive") &&
+      (otherLabels.findGroup("Archive").keywords() > 0)) {
+    outCube->putGroup(otherLabels.findGroup("Archive"));
+  }
+
+  if (otherLabels.hasGroup("Kernels") &&
+      (otherLabels.findGroup("Kernels").keywords() > 0)) {
+    outCube->putGroup(otherLabels.findGroup("Kernels"));
+  }
+
+
+
+  p.EndProcess();
+
+}
+
+
+/**
+ * This function reads Mars Express HRSC Stereo files
+ *
+ * First pass through the file is a Process called with the IgnoreData() function in order to
+ * get the import class to collect prefix data. Once the prefix data is populated, we look for
+ * "gaps" - HRSC files with prefix data give us a time and exposure duration for each line, we
+ * look for where the time + exposure duration != next line's time. We populate a table
+ * (LineScanTimes) with the prefix data and lineInFile with whether or   not a gap should be
+ * inserted.                                                            
+ *                                                                      
+ * Second pass through the file is a Process called with the WriteOutp  ut function. If there were
```

Comment:
Kind of weird additional spaces in this paragraph: `WriteOutp ut` for example

Reply:
fixed